### PR TITLE
feat: Add support to handle bytes for vector retrieval

### PR DESCRIPTION
**Pull request by Asher-hss**
## Description

Update retriever IO[bytes] support

## Motivation and Context

enhance retriever function #1001 

- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/retrievers/vector_retriever.py [hunk 1]
```diff
@@ -92,6 +92,14 @@ def process(
         """
         if isinstance(content, Element):
             elements = [content]
+        elif isinstance(content, IO):
+            from unstructured.partition.auto import partition
+
```
Comment (Wendong-Fan) [inline camel/retrievers/vector_retriever.py:97]: I think we need to use unstructured io by `from camel.loaders import UnstructuredIO
`, maybe we also need to do some polish in `UnstructuredIO`

File: camel/retrievers/vector_retriever.py [hunk 2]
```diff
@@ -72,7 +72,7 @@ def __init__(
 
     def process(
         self,
-        content: Union[str, Element],
+        content: Union[str, Element, IO[bytes]],
         chunk_type: str = "chunk_by_title",
         max_characters: int = 500,
```
Comment (Wendong-Fan) [inline camel/retrievers/vector_retriever.py:77]: for the chunk part, add one more argument to let user decide whether to chunk the content

**Review by Wendong-Fan — APPROVED** [review]
Thanks @Asher-hss ! 
Added one more commit here: https://github.com/camel-ai/camel/pull/1003/commits/b3d165dae453b37319a89c2122ec13a5ed715543
to add batch setting, for the PR naming please refer to our contribution guideline to make it more clear

File: camel/loaders/unstructured_io.py [hunk 3]
```diff
@@ -152,6 +153,41 @@ def parse_file_or_url(
                 warnings.warn(f"Failed to partition the file: {input_path}")
                 return None
 
+    @staticmethod
+    def parse_bytes(
+        file: IO[bytes], **kwargs: Any
```
Comment (koch3092) [inline camel/loaders/unstructured_io.py:158]: `**kwargs` itself inherently means `Any`, so there's no need to restrict it to the `Any `type.

**Review by koch3092 — COMMENTED** [review]
Thanks @Asher-hss, left a comment below.

File: camel/loaders/unstructured_io.py (hunk 3 above)
Comment (Asher-hss) [reply camel/loaders/unstructured_io.py:158]: Thanks, Coco. This part follows the format of the parse_file_or_url function.

**Review by koch3092 — APPROVED** [review]

**Pull request merged**